hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/DotNode.java

```diff
@@ -401,8 +402,14 @@ private void dereferenceEntity(
 			property = parentAsDotNode.propertyName;
 			joinIsNeeded = generateJoin && (
 					entityType.isNullable() ||
-					!isPropertyEmbeddedInJoinProperties( parentAsDotNode.propertyName )
+							!isPropertyEmbeddedInJoinProperties( parentAsDotNode.propertyName )
 			);
+			if ( generateJoin
+					&& implicitJoin
+					&& entityType.isNullable()
+					&& isReferenceToPrimaryKey( parentAsDotNode.propertyName, entityType ) ) {
```

Comment:
Trying to understand why `isReferenceToPrimaryKey` is needed here.  Given the other conditions, don't we always want to use LEFT?

Specifically, doesn't is-nullable itself indicate we need a left-join?

Reply:
Specifically, I mean isn't `generate && implicitJoin && entityType.isNullable()` enough?

Reply:
i think it can cause issues for some queries, I'm going to add a test for this